adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/PropertyBag.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.adapter.mqtt;
+
+import io.netty.handler.codec.http.QueryStringDecoder;
+import org.eclipse.hono.util.Constants;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * A collection of methods for processing <em>property-bag</em> set at the end of a topic.
+ *
+ */
+public final class PropertyBag {
+
+    private final String topic;
+
+    private PropertyBag(final String topic) {
+        this.topic = topic;
```

Comment:
`PropertyBag` seems to be an immutable. Wouldn't it therefore make sense to parse the properties contained in the topic once here, instead of parsing the properties every time you want to retrieve a value?

Reply:
@sophokles73 I have pushed the corresponding changes. Please have a look.